### DISCUSSION TITLE
GitHub deployment notifications

### DISCUSSION THREAD

**Discussion by hazcod** [fluxcd/flux2#2237] [Proposals] [answered]
It would be nice to be able to set deployment statuses on GitHub to mark it as success or failed.

**Comment by PatrykGala** [comment]
You can configure notifications to Github - https://fluxcd.io/docs/guides/notifications/

I don't see a integration with Github Deployment Env, maybe it is feature to implement. - [deployments#create-a-deployment-status](https://docs.github.com/en/rest/reference/deployments#create-a-deployment-status)

**Reply by GregoireW** [reply]
I confirm I had to build my own controller to be able to push kubernetes deployment status  to  github deployment.

I created my own controller because I needed a setup to link between a kubernetes object status and a github repository,  be able to extract data from somewhere (for instance the ref on which you create your deployment, I extract this from the image tag) and store some parameter.  in short it feels like this work a little bit complex for a simple notification process.

**Reply by kingdonb** [reply]
@GregoireW I saw you mentioned that you have built a controller to push to kubernetes deployment statuses!

It sounds like you're not happy with that, I agree that it should be supported as a part of Flux. Is your controller published somewhere though, so if there were any interesting challenges to solve, we can see how you solved them there? It might be helpful for the person to implement this in Flux (or maybe you're already working on this now. 👍 )

**Reply by GregoireW** [reply]
@kingdonb   I'm not really fully happy with what I did, and today it is not something I can opensource. ( When this is better, I can ask if it is possible, but not yet).

Now to send deployment info, you need multiple data like:  Which object to track, what the status, which repository to send status to, which environment it is, which ref it is, (and deal with github authentication).  
Basically, flux only know part of the status. And truth is a controller can live its own live without knowing anything about flux or CD part. So a question can be is this should be in flux or completely independent from flux.

**Comment by kingdonb** [comment]
This is a related discussion: https://github.com/fluxcd/flux2/discussions/1624

**Comment by LockedThread** [comment]
Bumping for exposure, this would be huge.

**Reply by stefanprodan** [reply]
See https://github.com/fluxcd/flux2/discussions/2237#discussioncomment-9041647

**Comment by GregoireW** [comment]
I may have forgotten to write a comment here.  If you want to check  https://github.com/decathlon/kubernetes-status-to-github/

**Comment by stefanprodan** [comment, marked as answer]
Flux supports GitHub `repository_dispatch` to trigger GH Actions, from an action you can update the deployment status https://fluxcd.io/flux/components/notification/providers/#setting-up-the-github-dispatch-provider

Example: https://fluxcd.io/flux/use-cases/gh-actions-helm-promotion/#define-the-promotion-github-workflow
